src/BenchmarkDotNet/Toolchains/MonoWasm/WasmAppSettings.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using BenchmarkDotNet.Toolchains.DotNetCli;
+using JetBrains.Annotations;
+
+
+namespace BenchmarkDotNet.Toolchains.MonoWasm
+{
+    [PublicAPI]
+    public class WasmAppSettings : NetCoreAppSettings
```

Comment:
I think that instead of introducing this type and `WasmSettings`, you can revert all the changes done to `NetCoreAppSettings`, remove `WasmAppSettings`, and expect all the necessary methods to just accept an instance of `NetCoreAppSettings` and an instance of `WasmSettings`. It should simplify the code